yt/data_objects/tests/test_dataset_access.py

```diff
@@ -95,6 +108,19 @@ def test_point_from_r():
     pt2 = ds.point([0.5,0.3,0.1])
     assert_equal(pt1["density"], pt2["density"])
 
+    # Test YTDimensionalityError
+    with assert_raises(YTDimensionalityError) as ex:
+        ds.r[0.5, 0.1]
+    assert_equal(str(ex.exception),
```

Comment:
Just a minor nit, but I'd fit the `assert_equal` within the `with` statement, so that it is obvious that the `ex` comes from the context.